graveler/graveler.go

```diff
@@ -1327,3 +1358,23 @@ func (g *Graveler) getCommitsForMerge(ctx context.Context, repositoryID Reposito
 	}
 	return fromCommit, toCommit, baseCommit, nil
 }
+
+func (g *Graveler) callPreCommitHooks(ctx context.Context, repositoryID RepositoryID, branchID BranchID, commit Commit) error {
+	for _, hook := range g.hooks.PreCommit {
```

Comment:
assuming each hook is an independent unit that doesn't depend on previous hooks, not sure we want to run them serially.. I'm usually opposed to doing concurrency where it's not required, but I'm not sure it's not required :) we already provide a pretty narrow timeframe for hooks to complete in - this makes the situation worse by having all hooks compete for a slice of that time..

Reply:
At graveler level, I don't have a problem running the hook one by one and keep it simple at this point.
Currently, we have only one consumer - entry catalog, and the execution of the webhooks will be in parallel.

Reply:
Moving the handling to EntryLevel and keeping simple callback at graveler level.